packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -363,7 +366,13 @@ contract BridgeFacet is BaseConnextFacet {
     {
       // Check that the asset is supported -- can be either adopted or local.
       TokenId memory canonical = s.adoptedToCanonical[_args.transactingAsset];
-      if (canonical.id == bytes32(0)) {
+
```

Comment:
we should avoid the `canonical` lookup here entirely if `_args.transactingAsset == address(0)`

Reply:
so

```solidity
// empty token id
TokenId memory canonical;
if (_args.transactingAsset != address(0)) {
  // NOW we get the token id fr
  canonical = s.adoptedToCanonical[_args.transactingAsset];

  // do all these other checks that only matter if transactingAsset is specified anyway
  if (canonical.id == bytes32(0)) {

      if (s.tokenRegistry.isLocalOrigin(_args.transactingAsset)) {
        revert BridgeFacet__xcall_notSupportedAsset();
      }

      (uint32 canonicalDomain, bytes32 canonicalId) = s.tokenRegistry.getTokenId(_args.transactingAsset);
      canonical = TokenId(canonicalDomain, canonicalId);
    }
}

Reply:
yeah that works too, ill make the change